python/tests/unit/arcticdb/version_store/test_deletion.py

```diff
@@ -285,3 +285,28 @@ def test_delete_date_range_pickled_symbol(lmdb_version_store):
     assert lmdb_version_store.is_symbol_pickled(symbol)
     with pytest.raises(InternalException) as e_info:
         lmdb_version_store.delete(symbol, (idx[1], idx[2]))
+
+
+def test_write_batch_prune_previous_versions(lmdb_version_store):
+    """Should be able to write different size of batch of data."""
+    lib = lmdb_version_store
+    lib_tool = lib.library_tool()
+    sym1 = "test_symbol1"
+    sym2 = "test_symbol2"
+    df0 = pd.DataFrame({"col_0": ["a", "b"]}, index=pd.date_range("2000-01-01", periods=2))
+    df1 = pd.DataFrame({"col_0": ["c", "d"]}, index=pd.date_range("2000-01-03", periods=2))
+
+    lib.batch_write([sym1], [df0])
+    lib.batch_write([sym1], [df1], prune_previous_version=True)
+
+    assert len(lib.list_versions(sym1)) == 1
+    assert len(lib_tool.find_keys(KeyType.TABLE_INDEX)) == 1
+    assert len(lib_tool.find_keys(KeyType.TABLE_DATA)) == 1
+
+    lib.batch_write([sym1, sym2], [df0, df0])
+    lib.batch_write([sym1, sym2], [df1, df1], prune_previous_version=True)
+
+    assert len(lib.list_versions(sym1)) == 1
+    assert len(lib.list_versions(sym2)) == 1
+    assert len(lib_tool.find_keys(KeyType.TABLE_INDEX)) == 2
+    assert len(lib_tool.find_keys(KeyType.TABLE_DATA)) == 2
```

Comment:
As @jamesmunro points out, can we also assert that the structure of:

1. Version ref keys
2. Symbol list keys

Is as expected. We should do this more broadly, but no reason not to do it for this test.

Reply:
I have also tested it with the VERSION and SYMBOL_LIST keys